### DISCUSSION TITLE
Lesson 6，thirdweb deploy, API key problem.

### DISCUSSION THREAD

**Discussion by RaferGuo** [Cyfrin/foundry-full-course-cu#431] [Q&A]
at video:[(7:51:44) | ThirdWeb Deploy](https://youtu.be/umepbfKp5rI?t=28304)
when i  enter 
```
npx thirdweb deploy
``` 
it let me  enter my API secret key.  i create api key on 
https://thirdweb.com/dashboard/settings/api-keys  and no matter use api key or client id. it shows as  below.  can anyone help me what can i do.thx.
```
Please enter your API secret key, you can find or create it on https://thirdweb.com/create-api-key 
… Ps4oGzjoEKSS3azRLUH5_JxkO1KjHH3PN_SMYDr6w3hJ8f6u827XymrjIlUWw2hMk-FYFNnS6afcUODJR1OgMg
SyntaxError: Unexpected token '<', "<!DOCTYPE "... is not valid JSON
    at JSON.parse (<anonymous>)
    at Response.json (file:///home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/node-fetch/src/body.js:149:15)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Ke (/home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/thirdweb/dist/cli/index.js:6:1270)
    at async mp (/home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/thirdweb/dist/cli/index.js:6:913)
    at async Ye (/home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/thirdweb/dist/cli/index.js:6:452)
    at async Command.<anonymous> (/home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/thirdweb/dist/cli/index.js:434:3798)
    at async Command.parseAsync (/home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/commander/lib/command.js:935:5)
    at async jm (/home/rafer/.npm/_npx/8d2f9c029655e602/node_modules/thirdweb/dist/cli/index.js:434:7324)
```
api-keys page
![4](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/cdccc9d9-e1e7-4030-b11a-6d90cb2e679c)

**Comment by cromewar** [comment]
It's stating you are using a invalid syntax on your code, could you please share it here so we can check out?

**Reply by RaferGuo** [reply]
its my SimpleStorage.sol
```
// SPDX-License-Identifier: MIT

pragma solidity 0.8.19;

contract SimpleStorage {
    uint256 myFavoriteNumber;

    struct Person {
        uint256 favoriteNumber;
        string name;
    }
    // uint256[] public anArray;
    Person[] public listOfPeople;

    mapping(string => uint256) public nameToFavoriteNumber;

    function store(uint256 _favoriteNumber)  public virtual{
        myFavoriteNumber = _favoriteNumber;
    }

    function retrieve() public view returns (uint256) {
        return myFavoriteNumber;
    }

    function addPerson(string memory _name, uint256 _favoriteNumber) public {
        listOfPeople.push(Person(_favoriteNumber, _name));
        nameToFavoriteNumber[_name] = _favoriteNumber;
    }
}
```
do you need any other code?
![5](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/99c10f12-d23e-419b-bce6-40549cf944b3)
and actually，when i clone from github https://github.com/Cyfrin/foundry-simple-storage-f23/tree/main
its still tell me to enter my API secret key. so i spent couple of hours and try different way to solve it, but dont work.

**Reply by Marfuen** [reply]
@RaferGuo Can you try logging in independently by running `npx thirdweb login` first -> paste in secret, and then try to deploy again?

**Comment by RaferGuo** [comment]
> @RaferGuo Can you try logging in independently by running `npx thirdweb login` first -> paste in secret, and then try to deploy again?
@Marfuen 
i did it，but its replay the similar content，and didn`t work.
![5](https://github.com/Cyfrin/foundry-full-course-f23/assets/31519376/06ba02cd-7405-47ce-bf4f-269179e840fb)

**Reply by Marfuen** [reply]
Hey sorry, I didn't see this. Can you join our discord and we can chat there?
My discord username is: @marfuen

https://discord.gg/thirdweb

**Reply by Marfuen** [reply]
I'm not sure what's wrong here, I've tried it myself using your key and even doing a fresh install of the thirdweb cli, can you try running it with `npx thirdweb@latest login`?

**Comment by DEVELOPER22DEVASHISH** [comment]
same problem facing no solution found anywhere ..i think this is the problem by thirdweb

**Comment by WahabAfridi789** [comment]
I am facing a similar kind of problem. When I enter the "deploy" command, it asks for an API key, but it does not allow me to paste or type the API key. As a result, I am unable to deploy the contract now.

**Reply by theirrationalone** [reply]
Just copy the secret api key and right click(way to paste things into terminal) and enter.... 🚀🚀🚀🚀

**Reply by Prejithpcet** [reply]
Not happening

**Reply by theirrationalone** [reply]
> Not happening

So it might be a BUG | mistake in thirdweb's deploy POST request with data body, Where they are "JSONIFYING" user's(our) posting data TWICE Or it could be same on their backend too(response).

Smash up/spam up their help/query/email support. 😂😂😂

I'll appreciate it.

**Reply by asharsayeed** [reply]
Copy the key and just right click where it asks on the cli. It will paste the key but does not show it. Hit enter it will work for sure.

**Reply by stymm-1** [reply]
git bash worked for me.

**Comment by u00w** [comment]
Same issue here since friday....using just the Unity sdk.

Thought it was me, tried in Powershell and Cmd shell.

No luck.
